src/main/java/me/dinowernli/grpc/prometheus/Configuration.java

```diff
@@ -32,23 +37,36 @@ public static Configuration allMetrics() {
     return new Configuration(
         true /* isIncludeLatencyHistograms */,
         CollectorRegistry.defaultRegistry,
-        DEFAULT_LATENCY_BUCKETS);
+        DEFAULT_LATENCY_BUCKETS,
+        new ArrayList<>());
   }
 
   /**
    * Returns a copy {@link Configuration} with the difference that Prometheus metrics are
    * recorded using the supplied {@link CollectorRegistry}.
    */
   public Configuration withCollectorRegistry(CollectorRegistry collectorRegistry) {
-    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, latencyBuckets);
+    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, latencyBuckets, labelHeaders);
   }
 
   /**
    * Returns a copy {@link Configuration} with the difference that the latency histogram values are
    * recorded with the specified set of buckets.
    */
   public Configuration withLatencyBuckets(double[] buckets) {
-    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, buckets);
+    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, buckets, labelHeaders);
+  }
+
+  /**
+   * Returns a copy {@link Configuration} that recognizes the given list of header names and uses their value from
```

Comment:
Can we add some explicit documentation about the cases:

* Special symbols (partially covered by the example, but it would be nice to be explicit)
* What happens when a header is missing (filled with empty string)